src/components/tx/ApprovalEditor/ApprovalValueField.tsx

```diff
@@ -0,0 +1,159 @@
+import NumberField from '@/components/common/NumberField'
+import TokenIcon from '@/components/common/TokenIcon'
+import { shortenAddress } from '@/utils/formatters'
+import { UNLIMITED_APPROVAL_AMOUNT } from '@/utils/tokens'
+import { validateAmount, validateDecimalLength } from '@/utils/validation'
+import { Autocomplete, Box, type MenuItemProps, Typography, MenuItem } from '@mui/material'
+import { ethers } from 'ethers'
+import { useController, useFormContext } from 'react-hook-form'
+import type { ApprovalInfo } from '.'
+import css from './styles.module.css'
+
+const ApprovalOption = ({
+  menuItemProps,
+  value,
+  decimals,
+}: {
+  menuItemProps: MenuItemProps
+  value: string
+  decimals?: number
+}) => {
+  let label = value
+  if (ethers.utils.formatUnits(UNLIMITED_APPROVAL_AMOUNT, decimals) === value) {
+    label = 'Unlimited'
+  }
+  if ('0' === value) {
+    label = 'Revoke'
+  }
+
+  return (
+    <MenuItem key={value} {...menuItemProps}>
+      {label}
+    </MenuItem>
+  )
+}
+
+export const ApprovalValueField = ({
+  name,
+  readonly = false,
+  tx,
+}: {
+  name: string
+  readonly?: boolean
+  tx: ApprovalInfo
+}) => {
+  const { control } = useFormContext() || {}
+
+  const {
+    field: { ref, onBlur, onChange, value },
+    fieldState,
+  } = useController({
+    name,
+    control,
+    rules: {
+      required: true,
+      validate: (val) => {
+        const decimals = tx.tokenInfo?.decimals
+        return validateAmount(val, true) || validateDecimalLength(val, decimals)
+      },
+    },
+  })
+
+  const label = fieldState.error?.message || 'Token'
+
+  const readonlyValueLabel =
+    ethers.utils.formatUnits(UNLIMITED_APPROVAL_AMOUNT, tx.tokenInfo?.decimals) === value ? 'Unlimited' : value
+
+  const options = ['0', ethers.utils.formatUnits(UNLIMITED_APPROVAL_AMOUNT, tx.tokenInfo?.decimals)]
+
+  return readonly ? (
```

Comment:
Is it not possible to make the `Autocomplete` readonly? The two `NumberField`s are almost identical and `${name}_readonly` is not present in RHF.